index.js

```diff
@@ -995,6 +1003,7 @@ module.exports = class Hypercore extends EventEmitter {
   }
 
   async append (blocks, opts = {}) {
+    this.tracer.trace()
```

Comment:
move here https://github.com/holepunchto/hypercore/pull/465/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R1015 and pass the blocks when its guaranteed to be an array